packages/domain/lib/use_case/check_version_use_case.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:domain/base/extension/string_to_double_extension.dart';
+import 'package:domain/repository/app_version_repository.dart';
+import 'package:domain/use_case/sample_use_case/use_case_out.dart';
+
+enum TypeNotificationVersion {
+  outdatedVersion,
+  suitableVersion,
+}
+
+class CheckVersionUseCase extends UseCaseOut<Future<TypeNotificationVersion?>> {
+  final AppVersionRepository versionRepository;
+
+  CheckVersionUseCase(this.versionRepository);
+
+  @override
+  Future<TypeNotificationVersion?> call() async {
+    final versionCollection = await versionRepository.requestVersions();
+    final minVersion = versionCollection.minVersion.toIntVersionFormat;
+    final actualVersion = versionCollection.actualVersion.toIntVersionFormat;
+    final currentVersion = versionCollection.currentVersion.toIntVersionFormat;
+    if (minVersion > currentVersion) {
+      return TypeNotificationVersion.outdatedVersion;
+    } else if (actualVersion > currentVersion && minVersion <= currentVersion) {
+      return TypeNotificationVersion.suitableVersion;
+    } else {
+      return null;
```

Comment:
return some success state instead of null. it seems to be more consistent. since null value can confuse, that we received some error from server/something didn't parse/we don't have some versions